### ISSUE TITLE
Switch to QHull directly?

### ISSUE THREAD

**Issue by saschatimme** [davidavdav/CHull.jl#19]
Hey there,

I use this in one of my packages and this dependency makes a little bit trouble during the installation / first use. On two of my colleagues systems (both Linux) I had to update the scipy version to get this working since Julia choose their global Python toolchain.
To avoid these kinds of problems, maybe we should instead wrap qhull directly? Or do you have other ideas to get the installation process more stable?

**Comment by saschatimme** [comment]
Okay the scipy bindings to qhull are definitely more involved than I thought, so this is easier said than done 😅

**Comment by blegat** [comment]
I agree, it would be nice to access qhull directly without scipy. It currently use scipy as it was easier to start with, see [this discussion](https://groups.google.com/forum/#!topic/julia-users/e9m8t5W3TVs/discussion).

**Comment by stevengj** [comment]
This package calls Qhull directly:  https://github.com/gridap/MiniQhull.jl

Maybe it should be merged with or replace Qhull.jl at some point?

**Comment by stevengj** [comment]
Now that Qhull 8.0.999 has been released on Yggdrasil (https://github.com/JuliaPackaging/Yggdrasil/pull/3657) with the new accessors API (https://github.com/qhull/qhull/pull/89), it should be much easier to call libqhull_r directly (as also discussed in https://github.com/gridap/MiniQhull.jl/issues/5) without requiring a C compiler or building a wrapper (as in https://github.com/JuhaHeiskala/DirectQhull.jl).

My only question is where that development should take place.  Here?  MiniQhull (cc @fverdugo)?  DirectQhull (cc @JuhaHeiskala)?

**Comment by JuhaHeiskala** [comment]
Ok, I thought just to take the [qhull.pyx](https://github.com/scipy/scipy/blob/4aebbd953fec41b2f3ff3ebbeb2bf91e88d0da2d/scipy/spatial/qhull.pyx) from Scipy and translate it to Julia more or less directly.

**Comment by JuhaHeiskala** [comment]
I have an initial version of updated DirectQhull in [8e43cd0d](https://github.com/JuhaHeiskala/DirectQhull.jl/commit/8e43cd0dcbd868a473293ff8d697e821c4f56501). In case someone wants to take a look/provide feedback on the design.
There is one kludge/workaround maybe due to issue in Julia itself. (Julia crashes with code I thought should work.)

So far building convex hull is supported with DirectQhull.ConvexHull(points)
ConvexHull now provides most of the features that Scipy ConvexHull has, but not yet all.

**Comment by thchr** [comment]
> I have an initial version of updated DirectQhull in [8e43cd0d](https://github.com/JuhaHeiskala/DirectQhull.jl/commit/8e43cd0dcbd868a473293ff8d697e821c4f56501). In case someone wants to take a look/provide feedback on the design.

I wanted to try out your commit but I'm not able to load the package (tried on Linux and Windows):
```jl
pkg> add https://github.com/JuhaHeiskala/DirectQhull.jl#master

julia> using DirectQhull
ERROR: KeyError: key DirectQhull [c3f9d41a-afcb-471e-bc58-0b8d83bd86f4] not found
Stacktrace:
 [1] getindex
   @ ./dict.jl:482 [inlined]
 [2] root_module
   @ ./loading.jl:979 [inlined]
 [3] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:945
 [4] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:923
```
I wonder if there's something off in the Project.toml?

**Comment by JuhaHeiskala** [comment]
Hi,
Yes, probably the Project.toml uuid is not correct. I have not yet published the updated version in Julia package system, so that might be the reason. And I don't now know how to best publish in development package (dev command in package manager)
You should be able to just include the DirectQhull.jl source file in Julia to try it out though. You do need the master version of Qhull_jll that has the new accessor functions included.

Maybe then better to move further issues to under DirectQhull from here.  :)
